src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,75 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  onClose: () => void;
+}
+
+export const TodoModal: React.FC<Props> = React.memo(({ todo, onClose }) => {
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = todo;
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(userId)
+      .then(userFromServer => {
+        setUser(userFromServer);
+      });
+  }, []);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+      {!user
```

Comment:
we usually start with positive condition implementation - and only then negative. Conditions like that are easier to read for humans